src/client/datascience/jupyter/kernels/cellExecution.ts

```diff
@@ -118,7 +127,8 @@ export class CellExecution {
         private readonly applicationService: IApplicationShell,
         private readonly kernelConnection: Readonly<KernelConnectionMetadata>,
         disposables: IDisposableRegistry,
-        extensionContext: IExtensionContext
+        extensionContext: IExtensionContext,
+        private readonly controller: NotebookController
```

Comment:
It think this works fine, just wondering why not just look it up off the manager from the cell uri?

Reply:
Actually never mind, that's a bad idea. That could change, and the execution is specific to a given connection. Ignore my comment :).

Reply:
> execution is specific to a given connection. Ignore my comment :).

Yes. Initially I did what you suggested. But each kernel/execution is tied to a specific connection (hence specific controller). 

It was much easier to just pass this, rather than passing the manager all around and then getting the controller (also ended up ina cyclic reference issues)